### PR TITLE
9 add grpc to mobile

### PR DESCRIPTION
**Pull request by Nephim**

### CODE REVIEW

File: src/UICrafter.Client/Layout/MainLayout.razor [hunk 1]
```diff
@@ -1,4 +1,6 @@
 @using System.Text
+@using System.Text.Json
```
Comment (Nephim) [inline src/UICrafter.Client/Layout/MainLayout.razor]: Remove these

File: src/UICrafter.Core/Components/LoadMetadataComponent.razor [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    public Guid UserId { get; set; } = Guid.Empty;
+    private IList<AppView> appViews;
+
+    protected override async Task OnInitializedAsync()
+    {
+        try
+        {
+            // Simulate getting data from repository
+            appViews = await AppViewRepository.GetAppViewsByUserIdAsync(UserId);
+            //Task.Delay(1000).Wait(); // Simulate loading delay
```
Comment (Nephim) [inline src/UICrafter.Core/Components/LoadMetadataComponent.razor]: Remove comment + when you do Task.Delay it is await Task.Delay not .Wait()

File: src/UICrafter.Core/Components/LoadMetadataComponent.razor [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            if (appViews == null || !appViews.Any())
```
Comment (Nephim) [inline src/UICrafter.Core/Components/LoadMetadataComponent.razor]: null check is appViews is null

File: src/UICrafter.Core/Components/LoadMetadataComponent.razor [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            {
+                Debug.WriteLine("No AppViews found, displaying mock items.");
```
Comment (Nephim) [inline src/UICrafter.Core/Components/LoadMetadataComponent.razor]: Log.Debug

File: src/UICrafter.Core/Components/LoadMetadataComponent.razor [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                // Set empty list to display mock items
+                appViews = new List<AppView>();
+            }
+        }
+        catch (Exception ex)
+        {
+            Debug.WriteLine($"Error fetching AppViews: {ex.Message}");
```
Comment (Nephim) [inline src/UICrafter.Core/Components/LoadMetadataComponent.razor]: Log.Debug

File: src/UICrafter.Core/Components/LoadMetadataComponent.razor [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            appViews = new List<AppView>(); // Ensure mock items are shown on error
+        }
+    }
+
+    private void OnViewClicked(long viewId)
+    {
+        Debug.WriteLine("Clicked a real view item");
```
Comment (Nephim) [inline src/UICrafter.Core/Components/LoadMetadataComponent.razor]: Log

File: src/UICrafter.Core/Components/LoadMetadataComponent.razor [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+        // Navigate to the corresponding view
+        NavigationManager.NavigateTo($"/button-test?viewId={viewId}");
+    }
+
+    private void OnMockItemClicked()
+    {
+        Debug.WriteLine("Clicked mock item");
```
Comment (Nephim) [inline src/UICrafter.Core/Components/LoadMetadataComponent.razor]: Log

File: src/UICrafter.Core/Components/LoadMetadataComponent.razor [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+        // Navigate to /button-test/1 when a mock item is clicked
+        NavigationManager.NavigateTo("/button-test/1");
```
Comment (Nephim) [inline src/UICrafter.Core/Components/LoadMetadataComponent.razor:60]: If it is a component that whould navigate without anthing else. You should be using href with <a> or something like it.

File: src/UICrafter.Core/Components/PocButtonNoMistakes.razor [hunk 9]
```diff
@@ -0,0 +1,60 @@
+@using MudBlazor
```
Comment (Nephim) [inline src/UICrafter.Core/Components/PocButtonNoMistakes.razor]: Move usings to _Imports

File: src/UICrafter.Core/Components/PocButtonNoMistakes.razor [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+@using UICrafter.Core.Models
+@using System.Diagnostics
+@using Microsoft.AspNetCore.Components
+@using System.Net.Http
+@using System.Net.Http.Json
+@using UICrafter.Core.Utility
+
+@inject HttpClient Http
```
Comment (Nephim) [inline src/UICrafter.Core/Components/PocButtonNoMistakes.razor]: IHttpClientProvider

File: src/UICrafter.Core/Components/PocButtonNoMistakes.razor [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    private bool isDisabled = false;
+    private Color buttonColor = Color.Primary;
+
+    private async Task CallApi()
+    {
+        try
+        {
+            isDisabled = true;
+            var originalLabel = Button.Button.Label;
+            Button.Button.Label = "Loading...";
```
Comment (Nephim) [inline src/UICrafter.Core/Components/PocButtonNoMistakes.razor:26]: I am not fan of such a loading thing. it is better to replace the text with a loading indicator or something along those lines. Keep it for now but should be reworked.

File: src/UICrafter.Core/Components/PocButtonNoMistakes.razor [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+            Button.Button.Label = originalLabel;
+            buttonColor = Color.Success;
+            isDisabled = false;
+
+            // Notify subscribers
+            StupidStore.NotifySubscribers(Button.Guid, response);
+        }
+        catch (Exception ex)
+        {
+            Debug.WriteLine($"Error calling API for button {Button.Button.Label}: {ex.Message}");
```
Comment (Nephim) [inline src/UICrafter.Core/Components/PocButtonNoMistakes.razor]: log

File: src/UICrafter.Core/Components/PocTextBox.razor [hunk 13]
```diff
@@ -0,0 +1,153 @@
+@using MudBlazor
```
Comment (Nephim) [inline src/UICrafter.Core/Components/PocTextBox.razor]: imports

File: src/UICrafter.Core/Components/PocTextBox.razor [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+                    Textbox.Textbox.Content = description;
+                }
+                else
+                {
+                    Textbox.Textbox.Content = $"Field '{fieldName}' not found in JSON response.";
+                }
+            }
+
+
+            // if (jsonResponse.TryGetValue(Textbox.Textbox.JsonField, out var fieldValue))
```
Comment (Nephim) [inline src/UICrafter.Core/Components/PocTextBox.razor]: commented?

File: src/testlib/Areas/MyFeature/Pages/Page1.cshtml [hunk 15]
```diff
@@ -0,0 +1,13 @@
+﻿@page
```
Comment (Aagaard00) [inline src/testlib/Areas/MyFeature/Pages/Page1.cshtml]: Delete

File: src/testlib/Areas/MyFeature/Pages/Page1.cshtml.cs [hunk 16]
```diff
@@ -0,0 +1,11 @@
+﻿namespace testlib.MyFeature.Pages;
```
Comment (Aagaard00) [inline src/testlib/Areas/MyFeature/Pages/Page1.cshtml.cs]: Delete

File: src/testlib/testlib.csproj [hunk 17]
```diff
@@ -0,0 +1,16 @@
+<Project Sdk="Microsoft.NET.Sdk.Razor">
```
Comment (Aagaard00) [inline src/testlib/testlib.csproj]: Delete

File: src/UICrafter.Core/Components/PocTextBox.razor [hunk 18]
```diff
@@ -0,0 +1,153 @@
...
+        if (value is List<object> list)
+        {
+            var itemsAsString = list.Select(item => item.ToString());
+            return string.Join(" ", itemsAsString);
+        }
+        else if (value is Dictionary<string, object> dict)
+        {
+            return JsonSerializer.Serialize(dict);
+
+            //return JsonConvert.SerializeObject(dict);
```
Comment (Nephim) [inline src/UICrafter.Core/Components/PocTextBox.razor]: Delete

File: src/UICrafter.Mobile/Components/Pages/AppViewPOC.razor [hunk 19]
```diff
@@ -0,0 +1,143 @@
+@page "/button-test/{viewId:long}"
+@using UICrafter.Core.AppView
```
Comment (Nephim) [inline src/UICrafter.Mobile/Components/Pages/AppViewPOC.razor]: Most usings into _Imports

File: src/UICrafter.Mobile/MauiProgram.cs [hunk 20]
```diff
@@ -27,6 +27,9 @@ public static MauiApp CreateMauiApp()
 
 		builder.Services.AddMauiBlazorWebView();
 		builder.Services.AddMudServices();
+		builder.Services.AddScoped(sp =>
```
Comment (Nephim) [inline src/UICrafter.Mobile/MauiProgram.cs]: We should look at this

**Pull request merged**